### ISSUE TITLE
Can you add a custom chain on mobile ?

### ISSUE THREAD

**Issue by talha-trili** [reown-com/reown-dotnet#23]
This is for mobile platforms.

Is it possible to add custom chains ?

Below is an example implementation and it doesnt add custom chains.
_**On webgl however, it adds custom chains and overall handles network switches better.**_

```
var supportedChains = new List<Chain>
                      {
	                      customChain1,
	                      customChain2,
                      };

appKitConfig.supportedChains = supportedChains.ToArray();

await AppKit.InitializeAsync(appKitConfig);
await AppKit.ConnectorController.TryResumeSessionAsync();
AppKit.AccountConnected    += OnAccountConnected;
AppKit.AccountDisconnected += OnAccountDisconnected;
AppKit.AccountChanged      += OnAccountChanged;
```

**Comment by skibitsky** [comment]
Hello,

The chains you set in `supportedChains` array should be available across all platforms. [Here's a relevant section in the documentation.](https://docs.reown.com/appkit/unity/core/options#supported-chains)

Could you please share the code for `customChain1` and `customChain2`? Maybe there's something in there that prevents them from appearing on mobile.

**Comment by LukaszwBeamable** [comment]
@skibitsky do you have any possible release date for that switching chain fix?

**Comment by skibitsky** [comment]
@LukaszwBeamable I'm aiming for tomorrow!

**Comment by skibitsky** [comment]
@LukaszwBeamable @talha-trili the latest version [v1.1.2](https://github.com/reown-com/reown-dotnet/releases/tag/com.reown.appkit.unity%2F1.1.2) has a potential fix for this issue

**Comment by skibitsky** [comment]
I've reproduced the issue: some chains (including Etherlink) aren't getting added to MetaMask. If the chain is already in your wallet, switching between networks works fine.
I'm working on a solution, but for now you can either manually add Etherlink to MetaMask or use a different wallet.

**Comment by talha-trili** [comment]
Hi @skibitsky, any updates ?

**Comment by talha-trili** [comment]
Also with the etherlink custom chain setup, i get null reference error at AccountController.cs line 167 after connecting to mm mobile. I believe this is because it doesnt enforce the custom chain and just connects to any other chains like ethereum, poly and etc.
<img width="814" alt="image" src="https://github.com/user-attachments/assets/0daccd84-a560-4b2c-8a3d-00da498cb3a5" />

**Comment by skibitsky** [comment]
Hello @talha-trili,

Sorry for the delayed answer. The good news is that I've found the issue! In your network configuration set native token decimal value to 6, but it has to be 18.
```csharp
Chain ETHERLINK_TESTNET = new(
                              ChainConstants.Namespaces.Evm,
                              "128123",
                              "Etherlink Testnet",
                              new Currency("Tez", "XTZ", 18), // 👈 use 18 here
                              new BlockExplorer("Explorer", "https://testnet.explorer.etherlink.com/"),
                              "https://node.ghostnet.etherlink.com",
                              true,
                              "https://etherlink.com/opengraph-image.png?4dd162b94a289c06",
                              "etherlinkTestnet"
                             );
```
I'll work on improving the error messages so you could get more useful information from MetaMask app!

**Comment by skibitsky** [comment]
@talha-trili, yes, MetaMask always connects the chain that is active in the MetaMask app, regardless of your app configuration. However, when you switch to the Etherlink in the dapp, AppKit will try to add that chain to the MetaMask. Basically, the AppKit chain will take priority over whatever is active in MetaMask when you make requests, but not during connection.

The problem is that there's a bug in latest version of MetaMask that essentially breaks chain switching 😔 We're in touch with the MetaMask team, I'll keep you posted.

**Comment by talha-trili** [comment]
Hi @skibitsky , i would like to raise this issue again, sdk does not add etherlink chain automatically to trust wallet android and throws unsupported chain exception. If i add etherlink to trust wallet manually then i can connect.

Adding custom chains automatically through reown sdk is a must to have since this provides a superior UX for users.

**Comment by skibitsky** [comment]
To add a new chain from the dapp, the wallet must support `wallet_addEthereumChain` method. Unfortunately this isn't something we have control over. Trust wallet has to implement `wallet_addEthereumChain`.

**Comment by skibitsky** [comment]
Hey, @talha-trili, we have improved chain switching in v1.3.0, give it a try!
